moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -387,30 +334,33 @@ bool planning_pipeline::PlanningPipeline::generatePlan(const planning_scene::Pla
         RCLCPP_DEBUG(LOGGER, "Planned path was found to be valid when rechecked");
       contacts_publisher_->publish(arr);
     }
-  }
 
-  // display solution path if needed
-  if (display_computed_motion_plans_ && solved)
-  {
-    moveit_msgs::msg::DisplayTrajectory disp;
-    disp.model_id = robot_model_->getName();
-    disp.trajectory.resize(1);
-    res.trajectory->getRobotTrajectoryMsg(disp.trajectory[0]);
-    moveit::core::robotStateToRobotStateMsg(res.trajectory->getFirstWayPoint(), disp.trajectory_start);
-    display_path_publisher_->publish(disp);
+    // Optionally publish DisplayTrajectory msg
+    if (display_computed_motion_plans)
```

Comment:
Why remove the `&& solved`?

Reply:
Hard to see from the diff but I moved the whole thing into the scope of `if (solved && res.trajectory)` to get rid of the additional solved check

Reply:
That would do it! Lol